lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2320,6 +2324,38 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
                                   transValue(II->getArgOperand(1), BB)};
     return BM->addExtInst(STy, BM->getExtInstSetId(SPIRVEIS_OpenCL), ExtOp, Ops,
                           BB);
+  }
+    // trata
+  case Intrinsic::umin:
+  case Intrinsic::umax:
+  case Intrinsic::smin:
+  case Intrinsic::smax: {
+    Type *BoolTy = IntegerType::getInt1Ty(M->getContext());
+    SPIRVValue *FirstArgVal = transValue(II->getArgOperand(0), BB);
+    SPIRVValue *SecondArgVal = transValue(II->getArgOperand(1), BB);
+
+    Op OC;
+    switch (II->getIntrinsicID()) {
+    case (Intrinsic::smin):
+      OC = OpSLessThan;
+      break;
+    case (Intrinsic::smax):
+      OC = OpSGreaterThan;
+      break;
+    case (Intrinsic::umin):
+      OC = OpULessThan;
+      break;
+    default:
+      OC = OpUGreaterThan;
+      break;
+    }
```

Comment:
This switch on Intrinsic ID inside the switch on Intrinsic ID seems weird. I would suggest to add something like `getBuiltinIdForIntrinsic` function or extend it so it handles not only instructions from OCL extended set or handle each intrinsic separately.